ggshield/cmd/iac/scan/precommit.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+import click
+
+from ggshield.cmd.iac.scan.diff import display_iac_scan_diff_result, iac_scan_diff
+from ggshield.cmd.iac.scan.iac_scan_common_options import (
+    add_iac_scan_common_options,
+    all_option,
+    directory_argument,
+    update_context,
+)
+from ggshield.core.config.config import Config
+from ggshield.core.text_utils import display_warning
+from ggshield.iac.iac_scan_models import create_client_from_config
+
+
+@click.command()
+@add_iac_scan_common_options()
+@all_option
+@directory_argument
+@click.pass_context
+def scan_pre_commit_cmd(
+    ctx: click.Context,
+    exit_zero: bool,
+    minimum_severity: str,
+    ignore_policies: Sequence[str],
+    ignore_paths: Sequence[str],
+    all: bool,
+    directory: Optional[Path] = None,
+    **kwargs: Any,
+) -> int:
+    """
+    Scan as pre-commit for IaC vulnerabilities. By default, it will return vulnerabilities added in the commit.
+    """
+    display_warning(
+        "This feature is still in beta, its behavior may change in future versions."
+    )
+
+    # TODO: WIP
+
+    if directory is None:
+        directory = Path().resolve()
+    update_context(ctx, exit_zero, minimum_severity, ignore_policies, ignore_paths)
+
+    # TODO: remove this once the GGClient is updated with the new diff function
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    ref = "HEAD"
+    staged = True
```

Comment:
maybe inline those?

(this comment and the two previous ones apply to the other `scan_pre_*_cmd()` functions as well)